### ISSUE TITLE
Check runtime camera permission

### ISSUE THREAD

**Issue by Zjarr** [andreipfeiffer/react-native-webview-android-file-upload#14]
Amazing module,

I just found that it has an issue, what if I need camera permission for other task? When this permission is removed, other modules that require this permission stops working. And if I add it but hadn't allowed the use of the camera, the app just stops working.

Is there a workaround?

Thank you, btw.

**Comment by andreipfeiffer** [comment]
Hey @Zjarr do you refer to the explicit permission that the README says should be removed?
`<uses-permission android:name="android.permission.CAMERA" />`

As far as I know, that is an old permission.
Maybe, you could add this attribute `android:maxSdkVersion="XX"`, so in newer versions is it now used.
Also, for newer SDK versions, you should use the runtime permissions for Camera.
I'm sure there are workarounds.
Honestly, I only had to support SDK 23+.

**Comment by Zjarr** [comment]
Sorry my english was awful.

My problem is that I need the camera permission to take a profile picture, that's why I can't remove this in the manifest, but if I keep it, your module crashes the entire app if the permission hasn't been granted.

Maybe you can try adding a permission check before opening the camera and if it hasn't been granted just don't let the user use it.

**Comment by andreipfeiffer** [comment]
That might work. I'll look at it when I get some time.

**Comment by Zjarr** [comment]
I've fixed my problem by adding permission checking.

This method handles the permission response and lunches the camera or just returns to the webview.
```
private PermissionListener permissionListener = new PermissionListener() {
  public boolean onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
    switch (requestCode) {
      case REQUEST_CAMERA: {
        if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
          startCamera(MediaStore.ACTION_IMAGE_CAPTURE, "image-", ".jpg");
          break;
        } else {
          filePathCallback.onReceiveValue(null);
          break;
        }
      }
    }
    return false;
  }
};
```

Requests the permission based on the kind of activity using the permissionListener above
```
@TargetApi(Build.VERSION_CODES.M)
private void requestPermissions() {
  if (getCurrentActivity() instanceof ReactActivity) {
    ((ReactActivity) getCurrentActivity()).requestPermissions(new String[]{ Manifest.permission.CAMERA}, REQUEST_CAMERA, permissionListener);
  } else {
    ((PermissionAwareActivity) getCurrentActivity()).requestPermissions(new String[]{ Manifest.permission.CAMERA}, REQUEST_CAMERA, permissionListener);
  }
}
```

Checks if the permission has been granted
```
@TargetApi(Build.VERSION_CODES.M)
private boolean permissionsGranted() {
  return ActivityCompat.checkSelfPermission(getCurrentActivity(), Manifest.permission.CAMERA) == PackageManager.PERMISSION_GRANTED;
}
```

Launches the camera or requests for camera permission
```
private void startCamera(String intentType, String prefix, String suffix) {
  if (permissionsGranted()) {
    Intent intent = new Intent(intentType);
    try {
      String packageName = getReactApplicationContext().getPackageName();
      File capturedFile = createCapturedFile(prefix, suffix);
      outputFileUri = FileProvider.getUriForFile(getReactApplicationContext(), packageName+".fileprovider", capturedFile);
      intent.putExtra(MediaStore.EXTRA_OUTPUT, outputFileUri);
      getCurrentActivity().startActivityForResult(intent, REQUEST_CAMERA);
    } catch (IOException ex) {
      Log.e("CREATE FILE", "Error occurred while creating the File", ex);
    }
  } else {
    requestPermissions();
   }
}
```

PD: I haven't pushed a pull request since I touched the code to meet my needs. Maybe in a near future I can do it.

**Comment by andreipfeiffer** [comment]
Fixed in `v1.4.0`
